src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,173 @@
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private Node<T> next;
+        private T value;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> addedElement = new Node<>(null, value, null);
+        if (isEmpty()) {
+            first = addedElement;
+            last = addedElement;
+        } else {
+            addedElement.prev = last;
+            last.next = addedElement;
+            last = addedElement;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size() || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index
+                   + " out of List bounds with size " + size);
+        }
+        if (index == size()) {
+            add(value);
+            return;
+        }
+        Node<T> nextElement = findElement(index);
+        Node<T> addedElement = new Node<>(nextElement.prev, value, nextElement);
+        if (nextElement.prev != null) {
+            nextElement.prev.next = addedElement;
+        } else {
+            first = addedElement;
+        }
+        nextElement.prev = addedElement;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element: list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findElement(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentElement = findElement(index);
+        T changedValue = currentElement.value;
+        currentElement.value = value;
+        return changedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> deletedElement;
+        deletedElement = checkBorderElement(index);
+        if (deletedElement != null) {
+            size--;
+            return deletedElement.value;
+        }
+        Node<T> currentElement = findElement(index);
+        unLink(currentElement);
+        size--;
+        return currentElement.value;
     }
 
     @Override
     public boolean remove(T object) {
+        int i = 0;
+        Node<T> currentElement = first;
+        while (currentElement != null && !(object == null && currentElement.value == null
+                || object != null && object.equals(currentElement.value))) {
+            currentElement = currentElement.next;
+            i++;
+        }
+        if (i < size) {
+            Node<T> deletedElement = checkBorderElement(i);
```

Comment:
variable is called deletedElement, but method checkBorderElement. Other people would normally expect that this method just checks something
You should move logic from method checkBorderElement() to unlink(). In checkBorderElement() you are unlinking too, why not to combine them